java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -443,13 +909,14 @@ public void delete(final String rawPath, final MantaHttpHeaders requestHeaders)
      * @param rawPath  Path of the object you want to delete.
      * @param requestHeaders  requestHeaders HTTP headers to attach to request (may be null)
      * @param pruneDepth the number of parent directories to be deleted if empty.
-     * @throws IOException
+     * @throws IOException If an IO exception has occurred.
      */
     void delete(final String rawPath, final MantaHttpHeaders requestHeaders, final Integer pruneDepth)
             throws IOException {
         Validate.notBlank(rawPath, "rawPath must not be blank");
         String path = formatPath(rawPath);
-        if (pruneDepth == null || pruneDepth == DEFAULT_PRUNE_DEPTH) {
+        if (pruneDepth == null || pruneDepth == DEFAULT_PRUNE_DEPTH
```

Comment:
Using `startsWith` in a hot operation like delete may be inefficient, consider optimizations - but don't go too crazy.

Reply:
Noted and added an additional check where the use case is: we have the prune directory settings (`pruneDepth > 0 || pruneDepth == -1`) configured for `mantaClient` in buckets environment, we print out a warning suggesting to disable those settings.